Mathlib/Computability/Ackermann.lean

```diff
@@ -348,3 +349,35 @@ theorem not_primrec_ack_self : ¬Primrec fun n => ack n n := by
 /-- The Ackermann function is not primitive recursive. -/
 theorem not_primrec₂_ack : ¬Primrec₂ ack := fun h =>
   not_primrec_ack_self <| h.comp Primrec.id Primrec.id
+
+open _root_.Partrec (Code) in
+/-- The Ackermann function is computable. -/
+theorem computable₂_ack : Computable₂ ack := by
+  suffices ∃ (ackCode : ℕ → Code), Primrec ackCode ∧
```

Comment:
You're proving as an auxiliary result that every function `ack n` is primitive recursive, right? Can you separate this into its own theorem?